spring.core/src/test/java/com/epam/springcore/task/facade/GymFacadeTest.java

```diff
@@ -0,0 +1,133 @@
+package com.epam.springcore.task.facade;
+
+import com.epam.springcore.task.model.Trainee;
+import com.epam.springcore.task.model.Trainer;
+import com.epam.springcore.task.model.Training;
+import com.epam.springcore.task.service.TraineeService;
+import com.epam.springcore.task.service.TrainerService;
+import com.epam.springcore.task.service.TrainingService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class GymFacadeTest {
+
+    @Mock
+    private TraineeService traineeService;
+
+    @Mock
+    private TrainerService trainerService;
+
+    @Mock
+    private TrainingService trainingService;
+
+    @InjectMocks
+    private GymFacade gymFacade;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+    }
+
+    @Test
+    void getTraineeById_ShouldDelegateToTraineeService() {
+        Trainee trainee = new Trainee();
+        trainee.setTraineeId(1L);
+
+        when(traineeService.getById(1L)).thenReturn(Optional.of(trainee));
+
+        Optional<Trainee> result = gymFacade.traineeService().getById(1L);
+
+        verify(traineeService).getById(1L);
```

Comment:
I prefer to use a black box testing approach which means that verify() is not needed, as this approach only tests that given some input there will be some result, in such cases, tests will be clearer and more readable.
verify() should be used only when you test void methods.